Extensions/bezier/bezier.lisp

```diff
@@ -317,10 +317,19 @@ second curve point, yielding (200 50)."
 (defmethod segments ((design bezier-design))
   (%segments design))
 
+(defun sequence-type (seq)
+  (let ((type (type-of seq)))
+    (if (and (listp type)
+             (eq (car type) 'vector)
+             (not (equal (third type)
+                         (length seq))))
+        (car type)
+        type)))
+
```

Comment:
won't traversing the sequence nullify the effect of knowing the exact size of it anyway?

Maybe we could just always force on transform-region type `vector`

`(map 'vector … …)` ?